lisa/microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -323,28 +328,50 @@ def _check_driver_installed(node: Node, log: Logger) -> None:
 
     if not gpu.is_supported():
         raise SkippedException(f"GPU is not supported with distro {node.os.name}")
-    if ComputeSDK.AMD in gpu.get_supported_driver():
-        raise SkippedException("AMD vm sizes is not supported")
 
     if isinstance(node.os, (Suse, AlmaLinux, Oracle)):
         raise SkippedException(
             f"{node.os.name} doesn't support GPU driver installation."
         )
 
-    try:
-        nvidia_smi = node.tools[NvidiaSmi]
-
-        lspci_gpucount = gpu.get_gpu_count_with_lspci()
-        nvidiasmi_gpucount = nvidia_smi.get_gpu_count()
-        assert_that(lspci_gpucount).described_as(
-            f"GPU count from lspci {lspci_gpucount} not equal to "
-            f"count from nvidia-smi {nvidiasmi_gpucount}"
-        ).is_equal_to(nvidiasmi_gpucount)
-    except Exception as e:
-        raise LisaException(
-            f"Cannot find nvidia-smi, make sure the driver installed correctly. "
-            f"Inner exception: {e}"
-        )
+    lspci_gpucount = gpu.get_gpu_count_with_lspci()
+
+    # Get the driver class for the supported GPU type
+    driver_class = _get_driver_class(node)
+
+    # Create GPU driver instance using virtual tool pattern
+    gpu_driver = node.tools.create(GpuDriver, driver_class=driver_class)
```

Comment:
It should pass in the ComputeSDK type, and map to the class in the GpuDriver class.

Reply:
Shall I move the ComputeSDK enum definition to a new common file under lisa/? 
Possibly gpu_types.py, so that both feature and tools can access it?